### PR TITLE
feat(route): Add Meta's Threads

### PR DESCRIPTION
**Pull request by ninboy**
## 该 PR 相关 Issue / Involved Issue

Close #

## 完整路由地址 / Example for the Proposed Route(s)

```routes
/threads/zuck
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [x] `Puppeteer`

## 说明 / Note
Creating some a basic route to extract Thread feeds.
image

### CODE REVIEW

**Comment by ninboy** [comment]
No idea why Puppeteer is timing out in Node 20, this only has a very basic use of it.

File: lib/v2/threads/index.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+const makeHeader = (user, appId, lsd) => ({
+    Accept: 'application/json',
+    Host: 'www.threads.net',
+    'Content-Type': 'application/x-www-form-urlencoded',
+    Origin: 'https://www.threads.net',
+    Referer: `https://www.threads.net/@${user}`,
+    'User-Agent': 'Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko; compatible; Googlebot/2.1; +http://www.google.com/bot.html) Chrome/83.0.4103.122 Safari/537.36',
+    'X-FB-LSD': lsd,
+    'X-IG-App-ID': appId,
```
Comment (TonyRL) [inline lib/v2/threads/index.js:55]: Look like this is the Threads internal app id in Meta which I guess is fixed `238260118697367`

File: lib/v2/threads/index.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    const body = makeBody({
+        lsd,
+        variables: encodeURIComponent(JSON.stringify({ userID: userId })),
+        doc_id: options.replies ? REPLIES_QUERY : THREADS_QUERY,
+    });
+
+    const resp = await got.post(apiUrl, {
+        headers,
+        body,
+    });
```
Comment (TonyRL) [inline lib/v2/threads/index.js:186]: You can do
```js
    const resp = await got.post(apiUrl, {
        headers,
        form: {
            lsd,
            variables: JSON.stringify({ userID: userId }),
            doc_id: options.replies ? REPLIES_QUERY : THREADS_QUERY,
        },
    });
```
In this way, `makeBody` can be safely removed. The header field `content-type` will be `application/x-www-form-urlencoded` automatically which you can skip in `makeHeader`. `accept` can also be skipped.

File: lib/v2/threads/index.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+    ctx.state.json.request = {
+        headers: resp.request.options.headers,
+        body: resp.request.options.body,
+    };
+
+    const responseBody = JSON.parse(resp.body);
```
Comment (TonyRL) [inline lib/v2/threads/index.js:193]:
```suggestion
    const responseBody = resp.data;
```
`@/utils/got` will try to perform a JSON.parse and put it under .data
https://github.com/DIYgod/RSSHub/blob/b6e1cf22aca3aff02c9244e0f64a276373dd5aa2/lib/utils/got.js#L15-L19

File: lib/v2/threads/maintainer.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    '/:user?': ['ninboy'],
```
Comment (TonyRL) [inline lib/v2/threads/maintainer.js:2]:
```suggestion
    '/:user/:routeParams?': ['ninboy'],
```
https://github.com/DIYgod/RSSHub/blob/847f0bf087755d2c4b81b21aba445be5a40a269c/lib/v2/threads/router.js#L1-L3

File: lib/v2/threads/radar.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'threads.net': {
+        _name: 'Threads',
+        '.': [
+            {
+                title: 'Threads',
```
Comment (TonyRL) [inline lib/v2/threads/radar.js:6]:
```suggestion
                title: 'User timeline',
```

File: docs/en/social-media.md [hunk 6]
```diff
@@ -440,6 +440,30 @@ Due to Telegram restrictions, some channels involving pornography, copyright, an
 
 <RouteEn author="fengkx" example="/telegram/blog" path="/telegram/blog" />
 
+## Threads
+
+### User timeline
+
+<RouteEn author="ninboy" path="/threads/:user/:routeParams?" example="/threads/zuck" radar="1" rssbud="1"/>
+cd .
```
Comment (TonyRL) [inline docs/en/social-media.md:448]:
```suggestion
<RouteEn author="ninboy" path="/threads/:user/:routeParams?" example="/threads/zuck" radar="1" rssbud="1">

```

File: docs/en/social-media.md [hunk 7]
```diff
@@ -440,6 +440,30 @@ Due to Telegram restrictions, some channels involving pornography, copyright, an
...
+| `showEmojiForQuotesAndReply`    | Use "🔁" instead of "QT", "↩️" instead of "Re"                                                                                 | `0`/`1`/`true`/`false` | `true`      |
+| `showQuotedInTitle`             | Show quoted tweet in title                                                                                                     | `0`/`1`/`true`/`false` | `true`      |
+| `replies`                       | Show replies                                                                                                                   | `0`/`1`/`true`/`false` | `true`      |
+
+Specify different option values than default values to improve readability. The URL
+
+```
+https://rsshub.app/threads/zuck/showAuthorInTitle=1&showAuthorInDesc=1&showQuotedAuthorAvatarInDesc=1&showAuthorAvatarInDesc=1&showEmojiForQuotesAndReply=1&showQuotedInTitle=1
+```
+
```
Comment (TonyRL) [inline docs/en/social-media.md:466]:
```suggestion

</RouteEn>

```

File: lib/v2/threads/index.js (hunk 2 above)
Comment (ninboy) [reply lib/v2/threads/index.js:186]: Thank you for such a detailed explanation

File: lib/v2/threads/radar.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'threads.net': {
+        _name: 'Threads',
+        '.': [
+            {
+                title: 'User timeline',
+                docs: 'https://docs.rsshub.app/social-media.html#threads',
+                source: ['/@:user'],
+                target: '/threads/:user',
```
Comment (TonyRL) [inline lib/v2/threads/radar.js:9]:
```suggestion
                source: ['/:user'],
                target: (params) => `/threads/${params.user.substring(1)}`,
```

File: lib/v2/threads/radar.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'threads.net': {
+        _name: 'Threads',
+        '.': [
+            {
+                title: 'User timeline',
+                docs: 'https://docs.rsshub.app/social-media.html#threads',
```
Comment (TonyRL) [inline lib/v2/threads/radar.js:7]:
```suggestion
                docs: 'https://docs.rsshub.app/en/social-media.html#threads',
```

**Pull request merged**